kura/org.eclipse.kura.wire.helper.provider/src/main/java/org/eclipse/kura/internal/wire/helper/WireHelperServiceImpl.java

```diff
@@ -113,16 +123,14 @@ public String getServicePid(final WireComponent wireComponent) {
     /** {@inheritDoc} */
     @Override
     public boolean isEmitter(final String wireComponentPid) {
-        requireNonNull(wireComponentPid, message.wireComponentPidNonNull());
+        requireNonNull(wireComponentPid, wireMessages.wireComponentPidNonNull());
         final BundleContext context = FrameworkUtil.getBundle(WireHelperServiceImpl.class).getBundleContext();
         final ServiceReference<?>[] refs = ServiceUtil.getServiceReferences(context, WireComponent.class, null);
         try {
             for (final ServiceReference<?> ref : refs) {
-                if (ref.getProperty(KURA_SERVICE_PID).equals(wireComponentPid)) {
-                    final WireComponent wc = (WireComponent) context.getService(ref);
-                    if (wc instanceof WireEmitter) {
-                        return true;
-                    }
+                final WireComponent wc = (WireComponent) context.getService(ref);
+                if (ref.getProperty(KURA_SERVICE_PID).equals(wireComponentPid) && wc instanceof WireEmitter) {
```

Comment:
getting service instance is more expensive than checking equals on string instance. It should be :

1. Check the property from the service reference
2. If true, get the service instance
3. Check if Emitter
4. If true, return the instance